### ISSUE TITLE
reportUnusedCallResult may trigger or not, depending on function name

### ISSUE THREAD

**Issue by abelcheung** [DetachHead/basedpyright#964]
[Minimal example](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMogCmAboQIYA2A%2BvAoVGQM4EwCwAUBwCaHCyGMYVVMEIgiXABQBKKAFoAfFAByYFIQBcHKDqgAPKAF4oARm268kvdI5A):

```python
from typing import reveal_type as rt

def test_inferred() -> None:
    x = 1
    rt(x)
```

The `reportUnusedCallResult` rule is triggered in this case, with exit code = 1. However, if I don't rename `reveal_type()`, then the rule is not triggered, and exit code = 0.

Actually `pyright` has same problem if I forcefully add `{"reportUnusedCallResult":"warning"}` to `pyrightconfig.json`. But since `basedpyright` opt to turn on this rule by default but not `pyright`, do you think if it worths fixing in `basedpyright`?

**Comment by abelcheung** [comment]
Hmm, on second thought, it should have been affected by the rule as well, but given how people commonly use `reveal_type()`, **only** this func [got special treatment](https://github.com/microsoft/pyright/blob/c9af81d96dcbf41d1949d2319ab4f088afc9bd12/packages/pyright-internal/src/analyzer/checker.ts#L809). What's your opinion here? I have disabled this rule locally in my tests, so it's fine either way.

**Comment by DetachHead** [comment]
yeah `reveal_type` is special cased here. related issues: #21 #544

**Comment by KotlinIsland** [comment]
also this:
```py
def reveal_type() -> int:
    return 1

reveal_type()  # no error

def foo() -> int:
    return 1

foo()  # error
```
[playground](https://basedpyright.com/?typeCheckingMode=all&reportUnusedParameter=true&reportRedeclaration=false&code=CYUwZgBATiBuIEMA2B9ALgTwA4gBQEoIBaAPggEsA7NALgFgAoCZ6ENAVykogEZHGY8ZOmx58-BqEhgA9jILEyVWoxasOXXhNnz8QA)
